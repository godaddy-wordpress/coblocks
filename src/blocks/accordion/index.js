/**
 * External dependencies
 */
import { AccordionIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	/* translators: block description */
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			count: 2,
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'tabs', 'coblocks' ),
		/* translators: block keyword (abbreviation for "frequently asked questions") */
		__( 'faq', 'coblocks' ),
	],
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	/* translators: block name */
	title: __( 'Accordion', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
