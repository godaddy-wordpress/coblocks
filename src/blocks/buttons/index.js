/**
 * THIS BLOCK IS DEPRECRATED. IT WILL NOT SHOW UP IN THE INSERTER ANYMORE.
 * SEE /src/js/deprecations/deprecate-coblocks-buttons.js
 */

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { button as icon } from '@wordpress/icons';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Buttons', 'coblocks' ),
	/* translators: block description */
	description: __( 'Prompt visitors to take action with multiple buttons, side by side.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'link', 'coblocks' ),
		/* translators: block keyword (abbreviation for "call to action") */
		__( 'cta', 'coblocks' ),
		/* translators: block keyword */
		__( 'call to action', 'coblocks' ),
	],
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
