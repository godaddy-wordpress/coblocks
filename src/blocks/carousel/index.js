/**
 * WordPress dependencies
 */
import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';

/**
 * Internal dependencies
 */
import BlockEdit from './block-edit';
import metadata from './block.json';

const settings = {
	edit: BlockEdit,
	save: () => (
		<div { ...useBlockProps.save() }>
			<InnerBlocks.Content />
		</div>
	),
};

const name = metadata.name;
export { name, metadata, settings };
