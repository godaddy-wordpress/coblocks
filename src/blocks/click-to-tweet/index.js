/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import { Icon } from '@wordpress/components';
import { TwitterIcon as icon } from '@godaddy-wordpress/coblocks-icons';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	deprecated,
	/* translators: block description */
	description: __( 'Add a quote for readers to tweet via Twitter.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			content: __( 'The easiest way to promote and advertise your blog, website, and business on Twitter.', 'coblocks' ),
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'share', 'coblocks' ),
		/* translators: block keyword */
		__( 'twitter', 'coblocks' ),
	],
	save,
	/* translators: block name */
	title: __( 'Click to Tweet', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
