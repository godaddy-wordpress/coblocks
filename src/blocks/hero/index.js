/**
 * External dependencies
 */
import { HeroIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon, Spinner } from '@wordpress/components';
import { lazy, Suspense } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { BackgroundAttributes } from '../../components/background';
import CSSGridAttributes from '../../components/grid-control/attributes';
import deprecated from './deprecated';
import DimensionsAttributes from '../../components/dimensions-control/attributes';
const Edit = lazy( () => import( './edit' ) );
import metadata from './block.json';
import ResponsiveBaseControlAttributes from '../../components/responsive-base-control/attributes';
import save from './save';
import transforms from './transforms';

/**
 * Block constants
 */
const { name, category } = metadata;

const attributes = {
	...CSSGridAttributes,
	...DimensionsAttributes,
	...BackgroundAttributes,
	...ResponsiveBaseControlAttributes,
	...metadata.attributes,
};

const settings = {
	attributes,
	deprecated,
	/* translators: block description */
	description: __( 'An introductory area of a page accompanied by a small amount of text and a call to action.', 'coblocks' ),
	edit: ( props ) => (
		<Suspense fallback={ <Spinner /> }>
			<Edit { ...props } />
		</Suspense>
	),
	example: {
		attributes: {},
		innerBlocks: [
			{
				attributes: {
					content: 'Welcome to CoBlocks',
				},
				name: 'core/heading',
			},
			{
				attributes: {
					content: 'CoBlocks is the most innovative collection of page building WordPress blocks for the new Gutenberg WordPress block editor.',
				},
				name: 'core/paragraph',
			},
			{
				attributes: {
					gutter: 'medium',
					items: 2,
				},
				innerBlocks: [
					{
						attributes: {
							text: 'See more',
						},
						name: 'core/button',
					},
					{
						attributes: {
							className: 'is-style-outline',
							text: 'Contact Us',
						},
						name: 'core/button',
					},
				],
				name: 'core/buttons',
			},
		],
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'button', 'coblocks' ),
		/* translators: block keyword (abbreviation for "call to action") */
		__( 'cta', 'coblocks' ),
		/* translators: block keyword */
		__( 'call to action', 'coblocks' ),
	],
	save,
	supports: {
		align: [ 'wide', 'full' ],
		coBlocksSpacing: true,
	},
	/* translators: block name */
	title: __( 'Hero', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings, attributes };
