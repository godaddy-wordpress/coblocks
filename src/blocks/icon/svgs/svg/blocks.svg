<svg fill="none" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="m10.3519669 11.7141378v8.2858622h8.2815735v-8.2858622zm-10.3519669 8.2858622h8.2815735v-8.2858622h-8.2815735zm0-18.64319006v8.28586225h8.2815735v-8.28586225zm14.1407867-1.35680994-5.8592132 5.85189021 5.8592132 5.86224759 5.8592133-5.86224759z" fill="currentColor" /></svg>