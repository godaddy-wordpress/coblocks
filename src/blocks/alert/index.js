/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const { replaceBlocks } = dispatch( 'core/block-editor' );

		replaceBlocks(
			[ props.clientId ],
			createBlock( 'core/paragraph', props.attributes ),
		);
		return null;
	},
	parent: [],
	save: () => null,
	title: metadata.title,
};

export { name, metadata, settings };
