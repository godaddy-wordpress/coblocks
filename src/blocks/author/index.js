/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch, select } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const parentBlock = select( 'core/editor' ).getBlocksByClientId( props.clientId )[ 0 ];
		if ( ! parentBlock ) {
			return null;
		}

		// Used to set text color props
		const hasTextColor = ( props.attributes.hasOwnProperty( 'textColor' ) );
		const hasCustomTextColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'text' ) );

		// Used to set background color props
		const hasBackgroundColor = ( props.attributes.hasOwnProperty( 'backgroundColor' ) );
		const hasCustomBackgroundColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'background' ) );

		// Used to set font props
		const hasFontSize = ( props.attributes.hasOwnProperty( 'fontSize' ) );
		const hasCustomFont = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'typography' ) );

		const { replaceBlocks } = dispatch( 'core/block-editor' );
		const isRTL = select( 'core/editor' ).getEditorSettings().isRTL;

		const authorNameBlock = () => {
			let authorNameBlockProps = {
				content: props.attributes.name,
				level: 4,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			// Headings do not inherit color from parent - we descend the style here for initial match.
			if ( hasCustomTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						color: { ...authorNameBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			return createBlock( 'core/paragraph', authorNameBlockProps );
		};

		const authorBioBlock = () => {
			let authorBioBlockProps = {
				content: props.attributes.biography,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					style: {
						...authorBioBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			return createBlock( 'core/paragraph', authorBioBlockProps );
		};

		const buttonBlock = parentBlock.innerBlocks;
		// Set buttons to align.
		buttonBlock.forEach( function( button, index ) {
			buttonBlock[ index ].attributes.align = isRTL ? 'right' : 'left';
		} );
		// Button blocks are no-longer stand-alone and should be within `core/buttons` block.
		const coreButtons = createBlock( 'core/buttons', { }, [ ...buttonBlock ] );

		const imageBlock = createBlock( 'core/image', { className: 'is-style-rounded', id: props.attributes.imgId, url: props.attributes.imgUrl } );

		const leftColumn = createBlock( 'core/column', { width: '25%' }, [ imageBlock ] );
		const rightColumn = createBlock( 'core/column', { width: '75%' }, [ authorNameBlock(), authorBioBlock(), coreButtons ] );

		const columnsBlock = () => {
			let columnsBlockProps = {
				...( props.attributes.hasOwnProperty( 'className' ) && { className: props.attributes.className } ),
				...( props.attributes.hasOwnProperty( 'animation' ) && { animation: props.attributes.animation } ),
				...( props.attributes.hasOwnProperty( 'anchor' ) && { anchor: props.attributes.anchor } ),
				style: {
					spacing: {
						padding: {
							bottom: '2.5rem',
							left: '2.5rem',
							right: '2.5rem',
							top: '2.5rem',
						},
					},
				},
			};

			// Migrate text colors into columns block props.
			if ( hasCustomTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			// Migrate background colors into columns block props.
			if ( hasCustomBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: props.attributes.style.color.background,
						},
					},
				};
			}
			if ( hasBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					backgroundColor: props.attributes.backgroundColor,
				};
			}

			// Does not have background color set. Use default block color from CoBlocks.
			const hasNoBackgroundColorSet = ! props.attributes.hasOwnProperty( 'backgroundColor' ) &&
				! props.attributes?.style?.color?.hasOwnProperty( 'background' );
			if ( hasNoBackgroundColorSet ) {
				// $dark-opacity-light-200 computed #8C8C971A - no variable access in root.
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: '#8C8C971A',
						},
					},
				};
			}

			return createBlock( 'core/columns', columnsBlockProps, [ leftColumn, rightColumn ] );
		};

		replaceBlocks(
			[ props.clientId ],
			columnsBlock()
		);

		return null;
	},
	parent: [],
	save: () => null,
	/* translators: block name */
	title: metadata.title,
};

export { name, metadata, settings };
