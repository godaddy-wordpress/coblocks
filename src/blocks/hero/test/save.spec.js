/**
 * External dependencies
 */
import { JSDOM } from 'jsdom';
import '@testing-library/jest-dom/extend-expect';
import { createBlock, registerBlockType, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let blockDOM;

describe( 'coblocks/hero', () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		blockDOM = undefined;
	} );

	it( 'should render', () => {
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should apply hero-center-center-align class if the "Center Center" layout option is selected', () => {
		block.attributes.layout = 'center-center';
		const serializedBlock = serialize( block );

		blockDOM = new JSDOM( serializedBlock );
		expect(
			blockDOM.window.document.querySelector( '.wp-block-coblocks-hero__inner' )
		).toHaveClass( `hero-${ block.attributes.layout }-align` );
	} );

	it( 'should apply has-no-padding class if the "none" padding setting is selected', () => {
		block.attributes.paddingSize = 'no';
		const serializedBlock = serialize( block );

		blockDOM = new JSDOM( serializedBlock );
		expect(
			blockDOM.window.document.querySelector( '.wp-block-coblocks-hero__inner' )
		).toHaveClass( 'has-no-padding' );
	} );

	it( 'should apply has-padding class if any padding setting is applied', () => {
		block.attributes.paddingSize = 'advanced';
		const serializedBlock = serialize( block );

		blockDOM = new JSDOM( serializedBlock );
		expect(
			blockDOM.window.document.querySelector( '.wp-block-coblocks-hero__inner' )
		).toHaveClass( 'has-padding' );
	} );

	it( 'should apply has-small-padding class if the small padding size is selected', () => {
		block.attributes.paddingSize = 'small';
		const serializedBlock = serialize( block );

		blockDOM = new JSDOM( serializedBlock );
		expect(
			blockDOM.window.document.querySelector( '.wp-block-coblocks-hero__inner' )
		).toHaveClass( `has-${ block.attributes.paddingSize }-padding` );
	} );

	it( 'should set content max width with inline css', () => {
		block.attributes.maxWidth = 750;
		const serializedBlock = serialize( block );

		blockDOM = new JSDOM( serializedBlock );
		expect(
			blockDOM.window.document.querySelector( '.wp-block-coblocks-hero__content' )
		).toHaveStyle( 'max-width: 750px' );
	} );

	it( 'should not set content max width when not defined', () => {
		block.attributes.maxWidth = null;
		const serializedBlock = serialize( block );

		expect( serializedBlock ).not.toContain( 'max-width' );
	} );

	it( 'should render with backgroundColor attribute', () => {
		block.attributes.backgroundColor = 'accent';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"backgroundColor":"accent"' );
		expect( serializedBlock ).toContain( 'has-accent-background' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with customBackgroundColor attribute', () => {
		block.attributes.customBackgroundColor = '#5a4a4f';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"customBackgroundColor":"#5a4a4f"' );
		expect( serializedBlock ).toContain( 'has-background' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with textColor attribute', () => {
		block.attributes.textColor = 'accent';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"textColor":"accent"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with customTextColor attribute', () => {
		block.attributes.customTextColor = '#5a4a4f';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"customTextColor":"#5a4a4f"' );
		expect( serializedBlock ).toContain( 'has-text-color' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with className attribute', () => {
		block.attributes.className = 'my-custom-class';
		const serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"className":"my-custom-class"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with contentAlign attribute', () => {
		const alignOptions = [ 'left', 'center', 'right' ];
		alignOptions.forEach( ( alignOption ) => {
			block.attributes.contentAlign = alignOption;
			const serializedBlock = serialize( block );

			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( `"contentAlign":"${ alignOption }"` );
			expect( serializedBlock ).toContain( `has-${ alignOption }-content` );
			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );

	it( 'should render with fullscreen & layout attributes', () => {
		const layoutOptions = [
			'top-left', 'top-center', 'top-right',
			'center-left', 'center-center', 'center-right',
			'bottom-left', 'bottom-center', 'bottom-right',
		];
		layoutOptions.forEach( ( layoutOption ) => {
			block.attributes.fullscreen = 'true';
			block.attributes.layout = layoutOption;
			const serializedBlock = serialize( block );
			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( '"fullscreen":"true"' );
			expect( serializedBlock ).toContain( 'is-fullscreen' );
			expect( serializedBlock ).toContain( `hero-${ layoutOption }-align` );

			if ( layoutOption !== 'center-left' ) { // center-left === default layout
				// eslint-disable-next-line jest/no-conditional-expect
				expect( serializedBlock ).toContain( `"layout":"${ layoutOption }"` );
			}
			expect( serializedBlock ).toMatchSnapshot();
		} );
	} );
} );
