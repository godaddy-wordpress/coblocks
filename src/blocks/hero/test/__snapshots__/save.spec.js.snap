// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/hero should render 1`] = `
"<!-- wp:coblocks/hero -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-center-left-align has-padding has-huge-padding" style="min-height:500px"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with backgroundColor attribute 1`] = `
"<!-- wp:coblocks/hero {"backgroundColor":"accent"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner has-accent-background-color has-background hero-center-left-align has-padding has-huge-padding has-accent-background-color" style="min-height:500px"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with className attribute 1`] = `
"<!-- wp:coblocks/hero {"className":"my-custom-class"} -->
<div class="wp-block-coblocks-hero alignfull my-custom-class"><div class="wp-block-coblocks-hero__inner hero-center-left-align has-padding has-huge-padding" style="min-height:500px"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with contentAlign attribute 1`] = `
"<!-- wp:coblocks/hero {"contentAlign":"left"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-center-left-align has-padding has-huge-padding has-left-content" style="min-height:500px"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with contentAlign attribute 2`] = `
"<!-- wp:coblocks/hero {"contentAlign":"center"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-center-left-align has-padding has-huge-padding has-center-content" style="min-height:500px"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with contentAlign attribute 3`] = `
"<!-- wp:coblocks/hero {"contentAlign":"right"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-center-left-align has-padding has-huge-padding has-right-content" style="min-height:500px"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with customBackgroundColor attribute 1`] = `
"<!-- wp:coblocks/hero {"customBackgroundColor":"#5a4a4f"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner has-background hero-center-left-align has-padding has-huge-padding" style="background-color:#5a4a4f;min-height:500px"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with customTextColor attribute 1`] = `
"<!-- wp:coblocks/hero {"customTextColor":"#5a4a4f"} -->
<div class="wp-block-coblocks-hero alignfull has-text-color" style="color:#5a4a4f"><div class="wp-block-coblocks-hero__inner hero-center-left-align has-padding has-huge-padding" style="min-height:500px"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with fullscreen & layout attributes 1`] = `
"<!-- wp:coblocks/hero {"layout":"top-left","fullscreen":"true"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-top-left-align has-padding has-huge-padding is-fullscreen"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with fullscreen & layout attributes 2`] = `
"<!-- wp:coblocks/hero {"layout":"top-center","fullscreen":"true"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-top-center-align has-padding has-huge-padding is-fullscreen"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with fullscreen & layout attributes 3`] = `
"<!-- wp:coblocks/hero {"layout":"top-right","fullscreen":"true"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-top-right-align has-padding has-huge-padding is-fullscreen"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with fullscreen & layout attributes 4`] = `
"<!-- wp:coblocks/hero {"fullscreen":"true"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-center-left-align has-padding has-huge-padding is-fullscreen"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with fullscreen & layout attributes 5`] = `
"<!-- wp:coblocks/hero {"layout":"center-center","fullscreen":"true"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-center-center-align has-padding has-huge-padding is-fullscreen"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with fullscreen & layout attributes 6`] = `
"<!-- wp:coblocks/hero {"layout":"center-right","fullscreen":"true"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-center-right-align has-padding has-huge-padding is-fullscreen"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with fullscreen & layout attributes 7`] = `
"<!-- wp:coblocks/hero {"layout":"bottom-left","fullscreen":"true"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-bottom-left-align has-padding has-huge-padding is-fullscreen"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with fullscreen & layout attributes 8`] = `
"<!-- wp:coblocks/hero {"layout":"bottom-center","fullscreen":"true"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-bottom-center-align has-padding has-huge-padding is-fullscreen"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with fullscreen & layout attributes 9`] = `
"<!-- wp:coblocks/hero {"layout":"bottom-right","fullscreen":"true"} -->
<div class="wp-block-coblocks-hero alignfull"><div class="wp-block-coblocks-hero__inner hero-bottom-right-align has-padding has-huge-padding is-fullscreen"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;

exports[`coblocks/hero should render with textColor attribute 1`] = `
"<!-- wp:coblocks/hero {"textColor":"accent"} -->
<div class="wp-block-coblocks-hero alignfull has-text-color has-accent-color"><div class="wp-block-coblocks-hero__inner hero-center-left-align has-padding has-huge-padding" style="min-height:500px"><div class="wp-block-coblocks-hero__content-wrapper"><div class="wp-block-coblocks-hero__content" style="max-width:560px"></div></div></div></div>
<!-- /wp:coblocks/hero -->"
`;
